src/main/java/pl/wojciechkostecki/revel/service/MenuService.java

```diff
@@ -0,0 +1,60 @@
+package pl.wojciechkostecki.revel.service;
+
+import org.springframework.stereotype.Service;
+import pl.wojciechkostecki.revel.exception.BadRequestException;
+import pl.wojciechkostecki.revel.mapper.MenuMapper;
+import pl.wojciechkostecki.revel.model.Local;
+import pl.wojciechkostecki.revel.model.Menu;
+import pl.wojciechkostecki.revel.model.dto.MenuDTO;
+import pl.wojciechkostecki.revel.repository.MenuRepository;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@Transactional
+public class MenuService {
+    private final MenuRepository menuRepository;
+    private final LocalService localService;
+    private final MenuMapper menuMapper;
+
+    public MenuService(MenuRepository menuRepository, LocalService localService, MenuMapper menuMapper) {
+        this.menuRepository = menuRepository;
+        this.localService = localService;
+        this.menuMapper = menuMapper;
+    }
+
+    public Menu save(MenuDTO menuDTO) {
+        Menu menu = menuMapper.toEntity(menuDTO);
+        Local local = localService.findById(menuDTO.getLocalId());
+        if (Objects.nonNull(local.getMenu())) {
+            throw new BadRequestException(local.getMenu().getId(), menuDTO.getLocalId());
+        }
+        menu.setLocal(local);
+        local.setMenu(menu);
+        return menuRepository.save(menu);
+    }
+
+    public List<Menu> getAll() {
+        return menuRepository.findAll();
+    }
+
+    public Menu findById(Long id) {
+        return menuRepository.findById(id)
```

Comment:
For these method its actually safer to return an optional and throw the exception where it is invoked. Sometimes you want to be able to give it different exception message.